main.cpp

```diff
@@ -0,0 +1,161 @@
+#include "ProductCatalog.h"
+#include "InventoryManager.h"
+#include "CustomExceptions.h"
+#include <iostream>
+#include <limits> // for clearing input buffer
+using namespace std;
+
+void clearInputBuffer() {
+    cin.clear();
+    cin.ignore(numeric_limits<streamsize>::max(), '\n');
+}
+
+void addProduct(ProductCatalog& catalog) {
+    int id;
+    string name;
+    float price;
+    int quantity;
+
+    cout << "Enter product ID: ";
+    if (!(cin >> id)) {

```

Comment:
I think it'll be better to make some type of validation functions as we are validating this type of stuff in multiple of the below functions as well.